plugins/utilities/mood_light.py

```diff
@@ -0,0 +1,248 @@
+                ba.textwidget(edit=self.lower_text, text=str(Ldefault))
+            elif self.selected == "upper":
+                Udefault -= 1
+                ba.textwidget(edit=self.upper_text, text=str(Udefault))
+        except AttributeError:
+            ba.textwidget(edit=self.warn_text, text="Click on number to select it")
+
+    def on_text_click(self, selected):
+        self.selected = selected
+        if selected == "upper":
+            ba.textwidget(edit=self.upper_text, color=(0, 0, 1))
+            ba.textwidget(edit=self.lower_text, color=(1, 1, 1))
+        elif selected == "lower":
+            ba.textwidget(edit=self.lower_text, color=(0, 0, 1))
+            ba.textwidget(edit=self.upper_text, color=(1, 1, 1))
+        else:
+            Print("this should't happen from on_text_click")
+
+    def draw_ui(self):
+        self.uiscale = ba.app.ui.uiscale
+
+        super().__init__(
+            root_widget=ba.containerwidget(
+                size=(670, 670),
+                on_outside_click_call=self.close,
+                transition="in_right",))
+
+        moodlight_label = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            position=(150, 550),
+            scale=2,
+            h_align="center",
+            v_align="center",
+            text="Mood light settings",
+            color=(0, 1, 0))
+
+        increase_button = ba.buttonwidget(
+            parent=self._root_widget,
+            position=(600, 100),
+            size=(5, 1),
+            scale=3.5,
+            extra_touch_border_scale=2.5,
+            icon=ba.gettexture("upButton"),
+            on_activate_call=self.increase_limit)
+
+        decrease_button = ba.buttonwidget(
+            parent=self._root_widget,
+            position=(100, 100),
+            size=(5, 1),
+            scale=3.5,
+            extra_touch_border_scale=2.5,
+            icon=ba.gettexture("downButton"),
+            on_activate_call=self.decrease_limit)
+
+        self.lower_text = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            scale=2,
+            position=(100, 200),
+            h_align="center",
+            v_align="center",
+            maxwidth=400.0,
+            text=str(Ldefault),
+            click_activate=True,
+            selectable=True)
+
+        lower_text_label = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            position=(100, 150),
+            h_align="center",
+            v_align="center",
+            text="Limit darkness")
+
+        self.upper_text = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            scale=2,
+            position=(400, 200),
+            h_align="center",
+            v_align="center",
+            maxwidth=400.0,
+            text=str(Udefault),
+            click_activate=True,
+            selectable=True)
```

Comment:
it's hard to navigate through these widgets on desktop version of the game, see if you can play around with their creation order, or [`up_widget`, `down_widget`, `left_widget`, `right_widget` params](https://docs.ballistica.net/api7/ba.html#widget) to make navigation seem more natural when using a keyboard; can also have this in future ^_^

Reply:
As a mobile player,i didn't know such issues existed.